pennylane/tape/tape.py

```diff
@@ -394,22 +394,17 @@ def _process_queue(self):
 
         Also calls `_update()` which sets many attributes.
         """
-        self._prep = []
-        self._ops = []
-        self._measurements = []
-        list_order = {"_prep": 0, "_ops": 1, "_measurements": 2}
-        current_list = "_prep"
-
-        for obj, info in self._queue.items():
-
-            if "owner" not in info and getattr(obj, "_queue_category", None) is not None:
-                if list_order[obj._queue_category] > list_order[current_list]:
-                    current_list = obj._queue_category
-                elif list_order[obj._queue_category] < list_order[current_list]:
-                    raise ValueError(
-                        f"{obj._queue_category[1:]} operation {obj} must occur prior "
-                        f"to {current_list[1:]}. Please place earlier in the queue."
-                    )
-                getattr(self, obj._queue_category).append(obj)
-
+        self._prep, self._ops, self._measurements = process_queue(self)
         self._update()
+
+    def __getitem__(self, key):
+        try:
+            return QuantumScript.__getitem__(self, key)
+        except TypeError:
```

Comment:
When is a `TypeError` raised?

Reply:
similar comment as for `__setitem__`, we're encapsulating two objects in one. If you do `tape[0]`, it should return the first element. If you pass too large of an index, you get an index error. But consider this:
```pycon
>>> with qml.tape.QuantumTape() as t:
...     op = qml.PauliX(0)
... 
>>> t[0]
PauliX(wires=[0])
>>> t[1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matthews/src/github.com/PennyLaneAI/pennylane/pennylane/tape/tape.py", line 402, in __getitem__
    return QuantumScript.__getitem__(self, key)
  File "/Users/matthews/src/github.com/PennyLaneAI/pennylane/pennylane/tape/qscript.py", line 214, in __getitem__
    return self.circuit[idx]
IndexError: list index out of range
>>> t[op]
{}
```

note that `t[op]` came from the AnnotatedQueue encapsulated within the QuantumTape. That's because it first tried to get it using `QuantumScript.__getitem__` (which is list indexing), got a `TypeError`, then tried the other way.

Reply:
Got it, thanks! I think this is worth commenting on in the code itself too.

Reply:
I've instead changed this to say `if isinstance(key, int)` then get from the `QuantumScript`. Otherwise get from the `AnnotatedQueue`. This avoids the try-except, and there's no mysterious `TypeError`